core/services/headtracker/head_saver_test.go

```diff
@@ -36,24 +90,20 @@ func Test_HeadSaver_addHeads(t *testing.T) {
 		}
 		parentHash = hash
 	}
-	headtracker.AddHeads(hs, heads, 6)
-	// Add duplicates (should be ignored)
-	headtracker.AddHeads(hs, heads[2:5], 6)
 
-	ch := hs.LatestChain()
-	assert.Equal(t, 6, len(headtracker.Heads(hs)))
+	// adding duplicates
+	heads = append(heads, heads[2:5]...)
+
+	for _, head := range heads {
+		err := saver.Save(context.TODO(), head)
+		require.NoError(t, err)
+	}
+
+	ch := saver.LatestChain()
 	require.NotNil(t, ch)
 	require.Equal(t, 5, int(ch.ChainLength()))
 
-	ch = hs.Chain(uncleHash)
-	assert.Equal(t, 6, len(headtracker.Heads(hs)))
+	ch = saver.Chain(uncleHash)
 	require.NotNil(t, ch)
 	require.Equal(t, 3, int(ch.ChainLength()))
```

Comment:
Why have the additional test cases been removed?

Reply:
Probably I simply forgot to restore them after the refactoring. Will take the second look.

Reply:
Fixed